### PR TITLE
Use write temp and replace to avoid 2 processes working on the same xlf file

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

File: src/XliffTasks/ExponentialRetry.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                Thread.Sleep(t);
+            }
+            throw new Exception("Timer should not be exhausted");
+        }
+
+        private static IEnumerable<TimeSpan> Intervals
+        {
+            get
+            {
+                var seconds = 5;
```
Comment (wli3) [inline src/XliffTasks/ExponentialRetry.cs]: 5 ms is enough

File: src/XliffTasks/Model/Document.cs [hunk 2]
```diff
@@ -62,12 +62,37 @@ public void Load(Stream stream)
...
             EnsureContent();
+            var tempPath = Path.Combine(Path.GetDirectoryName(path), Path.GetRandomFileName());
 
-            using (var stream = File.Open(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+            using (var stream = File.Open(tempPath, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
             {
                 Save(stream);
             }
+
+            ExponentialRetry.ExecuteWithRetryIOException(() =>
```
Comment (nguerrera) [inline src/XliffTasks/Model/Document.cs]: I'd add the word on between retry and IO to the method name

Comment (nguerrera) [reply src/XliffTasks/Model/Document.cs]: ExecuteWithRetryOnIOException

**Comment by wli3** [comment]
test ci please

**Comment by wli3** [comment]
@dotnet-bot test ci please

**Comment by wli3** [comment]
@dotnet-bot test this please

**Comment by wli3** [comment]
@dotnet-bot test ci please

**Comment by wli3** [comment]
@dontet-bot test ci please

**Comment by mmitche** [comment]
@wli3 The change will not take effect until after check-in...the netci.groovy file is not 'live' in this PR.

**Comment by wli3** [comment]
@mmitche thanks!

File: src/XliffTasks/ExponentialRetry.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            {
+                try
+                {
+                    action();
+                    return;
+                }
+                catch (IOException e)
+                {
+                    if (count == maxRetryCount)
+                        throw new AggregateException($"Retry failed after {count} times", e);
```
[the remark this answers is not in the document]
Comment (wli3) [reply src/XliffTasks/ExponentialRetry.cs]: fixed

File: src/XliffTasks/Model/Document.cs (hunk 2 above)
Comment (wli3) [reply src/XliffTasks/Model/Document.cs]: fixed

**Comment by wli3** [comment]
@dotnet-bot test ci please

**Comment by wli3** [comment]
Yay, CI passed, merging

**Pull request merged**